# IMDB Dataset - Assignment

### Question 1
Load the IMDb dataset using Pandas.

In [ ]:
# Write your solution here
import numpy as np
import pandas as pd

df = pd.read_csv('/content/imdb_data.csv')

### Question 2
Display the first 10 rows of the dataset.

In [ ]:
# Write your solution here
df.head(10)

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,2/20/15,93.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,8/6/04,113.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,10/10/14,105.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,3/9/12,122.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000
4,5,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt1380152,ko,마린보이,Marine Boy is the story of a former national s...,1.148070,...,2/5/09,118.0,"[{'iso_639_1': 'ko', 'name': '한국어/조선말'}]",Released,NaN,Marine Boy,NaN,"[{'cast_id': 3, 'character': 'Chun-soo', 'cred...","[{'credit_id': '52fe464b9251416c75073b43', 'de...",3923970
5,6,NaN,8000000,"[{'id': 16, 'name': 'Animation'}, {'id': 12, '...",NaN,tt0093743,en,Pinocchio and the Emperor of the Night,"Pinocchio and his friends, a glow worm and a m...",0.743274,...,8/6/87,83.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,NaN,Pinocchio and the Emperor of the Night,NaN,"[{'cast_id': 6, 'character': 'Pinocchio (voice...","[{'credit_id': '52fe46f49251416c9106558b', 'de...",3261638
6,7,NaN,14000000,"[{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...",http://www.thepossessionmovie.com/,tt0431021,en,The Possession,A young girl buys an antique box at a yard sal...,7.286477,...,8/30/12,92.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Fear The Demon That Doesn't Fear God,The Possession,NaN,"[{'cast_id': 23, 'character': 'Clyde', 'credit...","[{'credit_id': '52fe4981c3a368484e12ee29', 'de...",85446075
7,8,NaN,0,"[{'id': 99, 'name': 'Documentary'}]",NaN,tt0391024,en,Control Room,A chronicle which provides a rare window into ...,1.949044,...,1/15/04,84.0,"[{'iso_639_1': 'ar', 'name': 'العربية'}, {'iso...",Released,Different channels. Different truths.,Control Room,"[{'id': 917, 'name': 'journalism'}, {'id': 163...","[{'cast_id': 2, 'character': 'Himself', 'credi...","[{'credit_id': '52fe47a69251416c750a0daf', 'de...",2586511
8,9,"[{'id': 256377, 'name': 'The Muppet Collection...",0,"[{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...",NaN,tt0117110,en,Muppet Treasure Island,After telling the story of Flint's last journe...,6.902423,...,2/16/96,100.0,"[{'

### Question 3
Display the last 10 rows of the dataset.

In [ ]:
# Write your solution here
df.tail(10)

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
2990,2991,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...",NaN,tt0118663,en,B.A.P.S.,Can two clueless Georgia homegirls with big he...,1.560980,...,3/28/97,90.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,These Pretty Women... Are Clueless!,B.A.P.S.,"[{'id': 1437, 'name': 'burglar'}, {'id': 1946,...","[{'cast_id': 3, 'character': 'Nisi', 'credit_i...","[{'credit_id': '52fe477b9251416c7509af57', 'de...",7240837
2991,2992,"[{'id': 107469, 'name': 'Save The Last Dance C...",13000000,"[{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...",NaN,tt0206275,en,Save the Last Dance,"A white midwestern girl moves to Chicago, wher...",4.794088,...,1/12/01,112.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Only Person You Need To Be Is Yourself.,Save the Last Dance,"[{'id': 4118, 'name': 'ballet dancer'}, {'id':...","[{'cast_id': 18, 'character': 'Sara Johnson', ...","[{'credit_id': '52fe4533c3a36847f80c1c6b', 'de...",91038276
2992,2993,NaN,1135654,"[{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...",NaN,tt0015400,en,The Thief of Bagdad,A recalcitrant thief vies with a duplicitous M...,3.878515,...,3/18/24,149.0,"[{'iso_639_1': 'xx', 'name': 'No Language'}]",Released,"""Happiness Must Be Earned""",The Thief of Bagdad,"[{'id': 255, 'name': 'male nudity'}, {'id': 14...","[{'cast_id': 3, 'character': 'The Thief of Bag...","[{'credit_id': '52fe45bec3a368484e06c70b', 'de...",1213880
2993,2994,NaN,60000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",http://www.theterminal-themovie.com/,tt0362227,en,The Terminal,Viktor Navorski is a man without a country; hi...,14.092373,...,6/17/04,128.0,"[{'iso_639_1': 'bg', 'name': 'български език'}...",Released,Life is waiting.,The Terminal,"[{'id': 242, 'name': 'new york'}, {'id': 822, ...","[{'cast_id': 4, 'character': 'Viktor Navorski'...","[{'credit_id': '52fe4259c3a36847f8017699', 'de...",219417255
2994,2995,NaN,0,"[{'id': 18, 'name': 'Drama'}]",NaN,tt0105327,en,School Ties,When David Greene receives a football scholars...,7.438381,...,9/18/92,106.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Just because you’re accepted doesn’t mean you ...,School Ties,"[{'id': 6075, 'name': 'sport'}, {'id': 10144, ...","[{'cast_id': 2, 'character': 'David Greene', '...","[{'credit_id': '5637777ac3a3681b4d01f9f5', 'de...",14715067
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,4/22/94,102.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687
2996,2997,NaN,0,"[{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...",NaN,tt2364975,sv,Vi är bäst!,Three girls in 1980s Stockholm decide to form ...,3.727996,...,3/28/13,102.0,"[{'iso_639_1': 'sv', 'name': 'svenska'}]",Released,NaN,We Are the Best!,"[{'id': 1192, 'name': 'sweden'}, {'id': 4470, ...","[{'cast_id': 5, 'character': 'Bobo', 'credit_i...","[{'credit_id': '5716b72ac3a3686678012c84', 'de...",180590
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,10/11/96,120.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id': '52fe443a9251416c7502d579', 'de...",89456761
2998,2999,NaN,42000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",http://www.alongcamepolly.com/,tt0343135,en,Along Came Polly,Reuben Feffer is a 

### Question 4
Find the total number of rows and columns in the dataset.

In [ ]:
# Write your solution here
df.shape

(3000, 23)

### Question 5
Display all column names.

In [ ]:
# Write your solution here
df.columns

Index(['id', 'belongs_to_collection', 'budget', 'genres', 'homepage',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'runtime', 'spoken_languages',
       'status', 'tagline', 'title', 'Keywords', 'cast', 'crew', 'revenue'],
      dtype='object')

### Question 6
Check the data types of all columns.

In [ ]:
# Write your solution here
df.dtypes

,0
id,int64
belongs_to_collection,object
budget,int64
genres,object
homepage,object
imdb_id,object
original_language,object
original_title,object
overview,object
popularity,float64


### Question 7
Find the total number of missing values in each column.

In [ ]:
# Write your solution here
df.isnull().sum()

,0
id,0
belongs_to_collection,2396
budget,0
genres,7
homepage,2054
imdb_id,0
original_language,0
original_title,0
overview,8
popularity,0


### Question 8
Generate summary statistics using describe().

In [ ]:
# Write your solution here
df.describe()

,id,budget,popularity,runtime,revenue
count,3000.000000,3.000000e+03,3000.000000,2998.000000,3.000000e+03
mean,1500.500000,2.253133e+07,8.463274,107.856571,6.672585e+07
std,866.169729,3.702609e+07,12.104000,22.086434,1.375323e+08
min,1.000000,0.000000e+00,0.000001,0.000000,1.000000e+00
25%,750.750000,0.000000e+00,4.018053,94.000000,2.379808e+06
50%,1500.500000,8.000000e+06,7.374861,104.000000,1.680707e+07
75%,2250.250000,2.900000e+07,10.890983,118.000000,6.891920e+07
max,3000.000000,3.800000e+08,294.337037,338.000000,1.519558e+09


### Question 1
Display rows 10 to 20 and columns 2 to 5 using iloc.

In [ ]:
# Write your solution here
df.iloc[10:21,2:6]

,budget,genres,homepage,imdb_id
10,1000000,"[{'id': 18, 'name': 'Drama'}]",NaN,tt0075148
11,0,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt0093857
12,15000000,"[{'id': 18, 'name': 'Drama'}]",http://www.dreamworks.com/ab/,tt0169547
13,53000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 80, 'nam...",NaN,tt0377471
14,102000000,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt0181689
15,500000,"[{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...",http://skinningmovie.com/synopsis.htm,tt1129437
16,26000000,"[{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...",NaN,tt0421239
17,0,"[{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...",NaN,tt1700845
18,8000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt1487118
19,32000000,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt0388482


### Question 2
Display the first 15 rows using iloc.

In [ ]:
# Write your solution here
df.iloc[:16]

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,2/20/15,93.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,8/6/04,113.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,10/10/14,105.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,3/9/12,122.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000
4,5,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt1380152,ko,마린보이,Marine Boy is the story of a former national s...,1.148070,...,2/5/09,118.0,"[{'iso_639_1': 'ko', 'name': '한국어/조선말'}]",Released,NaN,Marine Boy,NaN,"[{'cast_id': 3, 'character': 'Chun-soo', 'cred...","[{'credit_id': '52fe464b9251416c75073b43', 'de...",3923970
5,6,NaN,8000000,"[{'id': 16, 'name': 'Animation'}, {'id': 12, '...",NaN,tt0093743,en,Pinocchio and the Emperor of the Night,"Pinocchio and his friends, a glow worm and a m...",0.743274,...,8/6/87,83.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,NaN,Pinocchio and the Emperor of the Night,NaN,"[{'cast_id': 6, 'character': 'Pinocchio (voice...","[{'credit_id': '52fe46f49251416c9106558b', 'de...",3261638
6,7,NaN,14000000,"[{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...",http://www.thepossessionmovie.com/,tt0431021,en,The Possession,A young girl buys an antique box at a yard sal...,7.286477,...,8/30/12,92.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Fear The Demon That Doesn't Fear God,The Possession,NaN,"[{'cast_id': 23, 'character': 'Clyde', 'credit...","[{'credit_id': '52fe4981c3a368484e12ee29', 'de...",85446075
7,8,NaN,0,"[{'id': 99, 'name': 'Documentary'}]",NaN,tt0391024,en,Control Room,A chronicle which provides a rare window into ...,1.949044,...,1/15/04,84.0,"[{'iso_639_1': 'ar', 'name': 'العربية'}, {'iso...",Released,Different channels. Different truths.,Control Room,"[{'id': 917, 'name': 'journalism'}, {'id': 163...","[{'cast_id': 2, 'character': 'Himself', 'credi...","[{'credit_id': '52fe47a69251416c750a0daf', 'de...",2586511
8,9,"[{'id': 256377, 'name': 'The Muppet Collection...",0,"[{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...",NaN,tt0117110,en,Muppet Treasure Island,After telling the story of Flint's last journe...,6.902423,...,2/16/96,100.0,"[{'

### Question 3
Extract only the columns: title, budget, revenue using loc.

In [ ]:
# Write your solution here
df.loc[:,['title','budget','revenue']]

,title,budget,revenue
0,Hot Tub Time Machine 2,14000000,12314651
1,The Princess Diaries 2: Royal Engagement,40000000,95149435
2,Whiplash,3300000,13092000
3,Kahaani,1200000,16000000
4,Marine Boy,0,3923970
...,...,...,...
2995,Chasers,0,1596687
2996,We Are the Best!,0,180590
2997,The Long Kiss Goodnight,65000000,89456761
2998,Along Came Polly,42000000,171963386


### Question 4
Display only the title and runtime columns.

In [ ]:
# Write your solution here
df.loc[:,['title','runtime']]

,title,runtime
0,Hot Tub Time Machine 2,93.0
1,The Princess Diaries 2: Royal Engagement,113.0
2,Whiplash,105.0
3,Kahaani,122.0
4,Marine Boy,118.0
...,...,...
2995,Chasers,102.0
2996,We Are the Best!,102.0
2997,The Long Kiss Goodnight,120.0
2998,Along Came Polly,90.0


### Question 5
Extract the last 20 rows of the dataset.

In [ ]:
# Write your solution here
df.iloc[-21:]

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
2979,2980,NaN,35000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",http://www.theback-upplan.com/,tt1212436,en,The Back-Up Plan,"When Zoe tires of looking for Mr. Right, she d...",6.629999,...,4/23/10,106.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,"Fall in love, get married, have a baby. Not ne...",The Back-Up Plan,"[{'id': 5091, 'name': 'kiss'}, {'id': 5098, 'n...","[{'cast_id': 22, 'character': 'Zoe', 'credit_i...","[{'credit_id': '52fe457e9251416c91034559', 'de...",77477008
2980,2981,NaN,300000,"[{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...",https://www.facebook.com/Actionherobiju/,tt5320514,ml,ആക്ഷൻ ഹീറോ ബിജു,Biju is an honest policeman who fights the inj...,0.350207,...,2/4/16,144.0,"[{'iso_639_1': 'ml', 'name': ''}]",Released,A ride with a police officer,Action Hero Biju,"[{'id': 1252, 'name': 'suicide attempt'}, {'id...","[{'cast_id': 1, 'character': 'Sub-Inspector Bi...","[{'credit_id': '56b4d47f92514112b90045f6', 'de...",2500000
2981,2982,NaN,0,"[{'id': 18, 'name': 'Drama'}]",NaN,tt4158876,en,Other People,"David, a struggling comedy writer fresh off fr...",3.069108,...,9/9/16,97.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,NaN,Other People,"[{'id': 234102, 'name': 'misfits'}]","[{'cast_id': 0, 'character': 'David', 'credit_...","[{'credit_id': '57d923309251415959005090', 'de...",93000
2982,2983,NaN,500000,"[{'id': 18, 'name': 'Drama'}]",NaN,tt2063781,en,Smashed,Kate and Charlie like to have a good time. The...,4.893812,...,9/12/12,85.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,NaN,Smashed,NaN,"[{'cast_id': 1001, 'character': 'Kate Hannah',...","[{'credit_id': '557455c1c3a3686b7d003df7', 'de...",376597
2983,2984,NaN,35000000,"[{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...",NaN,tt0118623,en,Anna Karenina,Anna (Marceau) is a wife and mother who has an...,5.660835,...,4/4/97,108.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,"In a world of power and privilege, one woman d...",Anna Karenina,NaN,"[{'cast_id': 1, 'character': 'Anna Karenina', ...","[{'credit_id': '52fe47cac3a36847f81484f7', 'de...",791830
2984,2985,"[{'id': 221111, 'name': 'S.W.A.T. Collection',...",80000000,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",http://www.sonypictures.com/homevideo/swat/,tt0257076,en,S.W.A.T.,Hondo Harrelson recruits Jim Street to join an...,13.334952,...,8/8/03,117.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,You're either S.W.A.T. or you're not.,S.W.A.T.,"[{'id': 2434, 'name': 'liberation'}, {'id': 35...","[{'cast_id': 25, 'character': ""Sgt. Dan 'Hondo...","[{'credit_id': '5379d4980e0a261431006f58', 'de...",116643346
2985,2986,NaN,37665000,"[{'id': 18, 'name': 'Drama'}, {'id': 53, 'name...",http://www.sonypictures.com/homevideo/freedoml...,tt0349467,en,Freedomland,A black police detective must solve a strange ...,5.388535,...,2/17/06,113.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,The Truth Is Hiding Where No One Dares To Look.,Freedomland,"[{'id': 577, 'name': 'black people'}, {'id': 5...","[{'cast_id': 13, 'character': 'Lorenzo Council...","[{'credit_id': '52fe4551c3a36847f80c76e7', 'de...",14655628
2986,2987,NaN,20000000,"[{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...",NaN,tt0338325,en,Paparazzi,A rising Hollywood actor decides to take perso...,12.553403,...,9/3/04,84.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,One good shot deserves another.,Paparazzi,"[{'id': 10183, 'name': 'independent film'}]","[{'cast_id': 1, 'character': 'Bo Laramie', 'cr...","[{'credit_id': '5678044292514179dd008c37', 'de...",16605763
2987,2988,NaN,0,"[{'id': 27, 'name': 'Horror'}, {'id': 878, 'na...",NaN,tt0096246,en,The Terror Within,Experiments in biological warfare have destroy...,1.292125,...,1/20/89,90.0,"[{'iso_639_1': 'en', 'name': 'E

### Question 6
Display rows 50 to 100 with only title and popularity columns.

In [ ]:
# Write your solution here
df.loc[50:101,['title','popularity']]

,title,popularity
50,"Moscow, Belgium",1.572904
51,The Great Race,4.473327
52,The Last Flight of Noah's Ark,1.012665
53,For Keeps,5.410839
54,Risen,8.759801
55,Son in Law,9.122718
56,Queen to Play,2.947980
57,Batman & Robin,17.038824
58,Were The World Mine,1.882446
59,All About the Benjamins,3.909079


### Question 1
Find all movies whose original language is English (en).

In [ ]:
# Write your solution here
df[df['original_language']=='en']

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,2/20/15,93.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,8/6/04,113.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,10/10/14,105.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000
5,6,NaN,8000000,"[{'id': 16, 'name': 'Animation'}, {'id': 12, '...",NaN,tt0093743,en,Pinocchio and the Emperor of the Night,"Pinocchio and his friends, a glow worm and a m...",0.743274,...,8/6/87,83.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,NaN,Pinocchio and the Emperor of the Night,NaN,"[{'cast_id': 6, 'character': 'Pinocchio (voice...","[{'credit_id': '52fe46f49251416c9106558b', 'de...",3261638
6,7,NaN,14000000,"[{'id': 27, 'name': 'Horror'}, {'id': 53, 'nam...",http://www.thepossessionmovie.com/,tt0431021,en,The Possession,A young girl buys an antique box at a yard sal...,7.286477,...,8/30/12,92.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Fear The Demon That Doesn't Fear God,The Possession,NaN,"[{'cast_id': 23, 'character': 'Clyde', 'credit...","[{'credit_id': '52fe4981c3a368484e12ee29', 'de...",85446075
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2994,2995,NaN,0,"[{'id': 18, 'name': 'Drama'}]",NaN,tt0105327,en,School Ties,When David Greene receives a football scholars...,7.438381,...,9/18/92,106.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Just because you’re accepted doesn’t mean you ...,School Ties,"[{'id': 6075, 'name': 'sport'}, {'id': 10144, ...","[{'cast_id': 2, 'character': 'David Greene', '...","[{'credit_id': '5637777ac3a3681b4d01f9f5', 'de...",14715067
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,4/22/94,102.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,10/11/96,120.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id': '52fe443a9251416c7502d579', 'de...",894

### Question 2
Find all movies whose original language is Hindi (hi).

In [ ]:
# Write your solution here
df[df['original_language']=='hi']

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,3/9/12,122.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000
213,214,NaN,6700000,"[{'id': 28, 'name': 'Action'}]",NaN,tt2077833,hi,राउडी राठौर,Low-life thief Shiva and top-cop Vikram Rathor...,1.312833,...,6/1/12,143.0,"[{'iso_639_1': 'hi', 'name': 'हिन्दी'}]",Released,NaN,Rowdy Rathore,"[{'id': 1721, 'name': 'fight'}, {'id': 8015, '...","[{'cast_id': 11, 'character': 'Shiva / Vikram ...","[{'credit_id': '52fe4a0bc3a36847f81b58a9', 'de...",31000000
371,372,NaN,13100000,"[{'id': 18, 'name': 'Drama'}, {'id': 35, 'name...",http://www.thankyouthefilm.com/,tt1720254,hi,Thank You,The film is about three husbands having affair...,1.417392,...,4/8/11,140.0,"[{'iso_639_1': 'hi', 'name': 'हिन्दी'}]",Released,Husbands will say sorry ... Wives will say tha...,Thank You,NaN,"[{'cast_id': 1, 'character': 'Kishen', 'credit...","[{'credit_id': '52fe4651c3a368484e08a13f', 'de...",4100000
401,402,NaN,3000000,"[{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...",NaN,tt0319736,hi,The Legend of Bhagat Singh,The Legend of Bhagat Singh is a 2002 Indian hi...,0.904409,...,6/7/02,155.0,"[{'iso_639_1': 'pa', 'name': 'ਪੰਜਾਬੀ'}, {'iso_...",Released,NaN,The Legend of Bhagat Singh,NaN,"[{'cast_id': 12, 'character': 'Bhagat K. Singh...","[{'credit_id': '52fe44849251416c9101372f', 'de...",6000000
419,420,NaN,3750000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10769, '...",http://www.atithithefilm.com/,tt1578116,hi,Atithi Tum Kab Jaoge,Atithi Tum Kab Jaoge? is the story of Puneet a...,0.597058,...,2/26/10,155.0,"[{'iso_639_1': 'hi', 'name': 'हिन्दी'}]",Released,NaN,"Dear Guest, When Will You Leave?",NaN,"[{'cast_id': 1, 'character': 'Puneet, Pappu', ...","[{'credit_id': '5647c0359251413ade004b4d', 'de...",5000000
481,482,NaN,10500000,"[{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...",http://www.cc2c-thefilm.com/,tt1091229,hi,Chandni Chowk To China,"Based in Delhi's Chandni Chowk, orphaned Sidhu...",1.970481,...,1/16/09,149.0,"[{'iso_639_1': 'de', 'name': 'Deutsch'}, {'iso...",Released,NaN,Chandni Chowk To China,"[{'id': 4344, 'name': 'musical'}, {'id': 10540...","[{'cast_id': 10, 'character': 'Sidhu', 'credit...","[{'credit_id': '52fe43d2c3a368484e00004d', 'de...",9400000
585,586,NaN,10400000,"[{'id': 18, 'name': 'Drama'}]",NaN,tt5074352,hi,दंगल,Dangal is an extraordinary true story based on...,8.792104,...,12/23/16,161.0,"[{'iso_639_1': 'hi', 'name': 'हिन्दी'}]",Released,NaN,Dangal,"[{'id': 5565, 'name': 'biography'}, {'id': 597...","[{'cast_id': 0, 'character': 'Mahavir Singh Ph...","[{'credit_id': '5714ef3f92514146c80003ea', 'de...",310000000
627,628,NaN,16000000,"[{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...",http://www.yashrajfilms.com/Movies/MovieIndivi...,tt3495026,hi,Fan,The story of a fan whose world completely revo...,5.127876,...,4/15/16,142.0,"[{'iso_639_1': 'hi', 'name': 'हिन्दी'}]",Released,World's biggest superstar got the universe's b...,Fan,"[{'id': 1523, 'name': 'obsession'}, {'id': 981...","[{'cast_id': 1, 'character': 'Gaurav Chandna /...","[{'credit_id': '569bc0de9251415e7000003c', 'de...",29251596
671,672,NaN,6400000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",http://www.anjaanaanjaani.erosentertainment.com,tt1499201,hi,Anjaana Anjaani,Akash (Ranbir Kapoor) and Kiara (Priyanka Chop...,1.822940,...,9/24/10,151.0,"[{'iso_639_1': 'hi', 'name': 'हिन्दी'}]",Released,All the greatest love stories...are between st...,Anjaana Anjaani,"[{'id': 1252, 'name': 'suicide attempt'}, {'id.

### Question 3
Display only the title and original_language columns for Hindi movies.

In [ ]:
# Write your solution here
df.loc[df['original_language']=='hi',['title','original_language']]

,title,original_language
3,Kahaani,hi
213,Rowdy Rathore,hi
371,Thank You,hi
401,The Legend of Bhagat Singh,hi
419,"Dear Guest, When Will You Leave?",hi
481,Chandni Chowk To China,hi
585,Dangal,hi
627,Fan,hi
671,Anjaana Anjaani,hi
695,Tere Naam,hi


### Question 4
Find movies having a budget greater than 100 million.

In [ ]:
# Write your solution here
df[df['budget']>100000000]

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
14,15,NaN,102000000,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt0181689,en,Minority Report,John Anderton is a top 'Precrime' cop in the l...,20.666063,...,6/20/02,145.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,The system is perfect until it comes after you.,Minority Report,"[{'id': 476, 'name': 'self-fulfilling prophecy...","[{'cast_id': 5, 'character': 'Chief John Ander...","[{'credit_id': '52fe4223c3a36847f8006f3d', 'de...",358372926
34,35,"[{'id': 90863, 'name': 'Rush Hour Collection',...",140000000,"[{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...",NaN,tt0293564,en,Rush Hour 3,After an attempted assassination on Ambassador...,9.718111,...,8/8/07,91.0,"[{'iso_639_1': 'la', 'name': 'Latin'}, {'iso_6...",Released,The Rush Is On!,Rush Hour 3,"[{'id': 1704, 'name': 'ambassador'}]","[{'cast_id': 2, 'character': 'Det. James Carte...","[{'credit_id': '52fe43fac3a36847f807b5bd', 'de...",258022233
57,58,"[{'id': 120794, 'name': 'Batman Collection', '...",125000000,"[{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...",NaN,tt0118688,en,Batman & Robin,Along with crime-fighting partner Robin and ne...,17.038824,...,6/20/97,125.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Strength. Courage. Honor. And loyalty.,Batman & Robin,"[{'id': 848, 'name': 'double life'}, {'id': 84...","[{'cast_id': 4, 'character': 'Bruce Wayne / Ba...","[{'credit_id': '59b66a169251417cbc011ec4', 'de...",238207122
60,61,"[{'id': 645, 'name': 'James Bond Collection', ...",150000000,"[{'id': 12, 'name': 'Adventure'}, {'id': 28, '...",http://www.mgm.com/#/our-titles/233/Casino-Roy...,tt0381061,en,Casino Royale,"Le Chiffre, a banker to the world's terrorists...",23.065078,...,11/14/06,144.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,Everyone has a past. Every legend has a beginn...,Casino Royale,"[{'id': 131, 'name': 'italy'}, {'id': 383, 'na...","[{'cast_id': 1, 'character': 'James Bond', 'cr...","[{'credit_id': '53ac1fd7c3a3684bc8001908', 'de...",599045960
121,122,"[{'id': 106768, 'name': 'Tarzan (Animation) Co...",150000000,"[{'id': 12, 'name': 'Adventure'}, {'id': 16, '...",http://disneydvd.disney.go.com/tarzanr-special...,tt0120855,en,Tarzan,Tarzan was a small orphan who was raised by an...,12.453452,...,6/18/99,88.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,An immortal legend. As you've only imagined.,Tarzan,"[{'id': 690, 'name': 'gorilla'}, {'id': 1009, ...","[{'cast_id': 9, 'character': 'Tarzan', 'credit...","[{'credit_id': '5912c9a19251414e8005140a', 'de...",448000000
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2866,2867,"[{'id': 52984, 'name': 'National Treasure Coll...",130000000,"[{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...",http://disney.go.com/disneypictures/nationaltr...,tt0465234,en,National Treasure: Book of Secrets,Benjamin Franklin Gates and Dr. Abigail Chase ...,12.854373,...,12/13/07,124.0,"[{'iso_639_1': 'la', 'name': 'Latin'}, {'iso_6...",Released,The Greatest Adventure History Has Ever Revealed,National Treasure: Book of Secrets,"[{'id': 90, 'name': 'paris'}, {'id': 212, 'nam...","[{'cast_id': 1, 'character': 'Benjamin Frankli...","[{'credit_id': '52fe4460c3a36847f8092639', 'de...",457363168
2880,2881,NaN,150000000,"[{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...",NaN,tt2334879,en,White House Down,Capitol Policeman John Cale has just been deni...,9.920223,...,6/27/13,131.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It will start like any other day.,White House Down,"[{'id': 840, 'name': 'usa president'}, {'id': ...","[{'cast_id': 9, 'character': 'Cale', 'credit_i...","[{'credit_id': '52fe4baec3a36847f820f3b1', 'de...",205366737
2899,2900,NaN,127500000,"[{'id': 16, 'name': 'Animation'}, {'id': 10751...",NaN,tt0130623,en,D

### Question 5
Find movies with revenue greater than 500 million.

In [ ]:
# Write your solution here
df[df['revenue']>500000000]

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
46,47,"[{'id': 8354, 'name': 'Ice Age Collection', 'p...",80000000,"[{'id': 16, 'name': 'Animation'}, {'id': 10751...",http://www.iceagemovies.com/films/ice-age-the-...,tt0438097,en,Ice Age: The Meltdown,"Diego, Manny and Sid return in this sequel to ...",16.646029,...,3/23/06,91.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Ice age is melting away.,Ice Age: The Meltdown,"[{'id': 2078, 'name': 'mammoth'}, {'id': 2079,...","[{'cast_id': 1, 'character': 'Manny (voice)', ...","[{'credit_id': '52fe4292c3a36847f80292c9', 'de...",660940780
60,61,"[{'id': 645, 'name': 'James Bond Collection', ...",150000000,"[{'id': 12, 'name': 'Adventure'}, {'id': 28, '...",http://www.mgm.com/#/our-titles/233/Casino-Roy...,tt0381061,en,Casino Royale,"Le Chiffre, a banker to the world's terrorists...",23.065078,...,11/14/06,144.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,Everyone has a past. Every legend has a beginn...,Casino Royale,"[{'id': 131, 'name': 'italy'}, {'id': 383, 'na...","[{'cast_id': 1, 'character': 'James Bond', 'cr...","[{'credit_id': '53ac1fd7c3a3684bc8001908', 'de...",599045960
114,115,"[{'id': 119, 'name': 'The Lord of the Rings Co...",93000000,"[{'id': 12, 'name': 'Adventure'}, {'id': 14, '...",http://www.lordoftherings.net/,tt0120737,en,The Lord of the Rings: The Fellowship of the Ring,"Young hobbit Frodo Baggins, after inheriting a...",32.070725,...,12/18/01,178.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,One ring to rule them all,The Lord of the Rings: The Fellowship of the Ring,"[{'id': 603, 'name': 'elves'}, {'id': 604, 'na...","[{'cast_id': 28, 'character': 'Frodo Baggins',...","[{'credit_id': '52fe421ac3a36847f80043ef', 'de...",871368364
188,189,NaN,180000000,"[{'id': 16, 'name': 'Animation'}, {'id': 10751...",http://disney.go.com/disneypictures/wall-e/,tt0910970,en,WALL·E,WALL·E is the last robot left on an Earth that...,16.088366,...,6/22/08,98.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,An adventure beyond the ordinar-E.,WALL·E,"[{'id': 9799, 'name': 'romantic comedy'}]","[{'cast_id': 12, 'character': 'WALL¬∑E / M-O (...","[{'credit_id': '52fe43a29251416c75018101', 'de...",521311860
203,204,NaN,75000000,"[{'id': 16, 'name': 'Animation'}, {'id': 35, '...",http://www.simpsonsmovie.com/,tt0462538,en,The Simpsons Movie,After Homer accidentally pollutes the town's w...,14.302911,...,7/25/07,87.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,See our family. And feel better about yours.,The Simpsons Movie,"[{'id': 494, 'name': 'father son relationship'...","[{'cast_id': 2, 'character': ""Homer / Itchy / ...","[{'credit_id': '52fe4211c3a36847f8001521', 'de...",527068851
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2647,2648,"[{'id': 87359, 'name': 'Mission: Impossible Co...",145000000,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",http://www.missionimpossible.com/,tt1229238,en,Mission: Impossible - Ghost Protocol,In the 4th installment of the Mission Impossib...,14.258430,...,12/7/11,133.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,No Plan. No Backup. No Choice.,Mission: Impossible - Ghost Protocol,"[{'id': 1721, 'name': 'fight'}, {'id': 9663, '...","[{'cast_id': 5, 'character': 'Ethan Hunt', 'cr...","[{'credit_id': '53563fb00e0a262861002dd0', 'de...",694713380
2737,2738,"[{'id': 645, 'name': 'James Bond Collection', ...",245000000,"[{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...",http://www.sonypictures.com/movies/spectre/,tt2379713,en,Spectre,A cryptic message from Bond‚Äôs past sends him...,24.926577,...,10/26/15,148.0,"[{'iso_639_1': 'fr', 'name': 'Français'}, {'is...",Released,A Plan No One Escapes,Spectre,"[{'id': 470, 'name': 'spy'}, {'id': 818, 'name...","[{'cast_id': 1, 'character': 'James Bond', 'cr

### Question 6
Find movies whose runtime is greater than 150 minutes.

In [ ]:
# Write your solution here
df[df['runtime']>150]

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
47,48,NaN,44000000,"[{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...",http://thehatefuleight.com/,tt3460252,en,The Hateful Eight,Bounty hunters seek shelter from a raging bliz...,20.328822,...,12/25/15,167.0,"[{'iso_639_1': 'es', 'name': 'Español'}, {'iso...",Released,No one comes up here without a damn good reason.,The Hateful Eight,"[{'id': 801, 'name': 'bounty hunter'}, {'id': ...","[{'cast_id': 22, 'character': 'Major Marquis W...","[{'credit_id': '538aa95ec3a368714e0005b3', 'de...",155760117
51,52,NaN,12000000,"[{'id': 12, 'name': 'Adventure'}, {'id': 35, '...",NaN,tt0059243,en,The Great Race,"Professional daredevil and white-suited hero, ...",4.473327,...,7/1/65,160.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,"The movie with 20,000-mile or one-million-laug...",The Great Race,"[{'id': 9253, 'name': 'slapstick'}, {'id': 159...","[{'cast_id': 1, 'character': 'Prof. Fate', 'cr...","[{'credit_id': '52fe445e9251416c75032581', 'de...",25333333
76,77,NaN,500000,"[{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...",NaN,tt0242572,ta,Kandukondain Kandukondain,Two young women with differing views on love f...,0.660663,...,5/5/00,157.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,NaN,Kandukondain Kandukondain,NaN,"[{'cast_id': 1, 'character': 'Manohar', 'credi...","[{'credit_id': '52fe47c7c3a368484e0d8f41', 'de...",500000
97,98,NaN,0,"[{'id': 14, 'name': 'Fantasy'}, {'id': 18, 'na...",NaN,tt1772980,ru,Mishen,"In the year 2020, a group of wealthy Moscovite...",1.042112,...,6/26/11,158.0,"[{'iso_639_1': 'de', 'name': 'Deutsch'}, {'iso...",Released,NaN,Target,NaN,"[{'cast_id': 1001, 'character': 'Victor', 'cre...","[{'credit_id': '58cdbad4c3a36850e9011085', 'de...",73000
107,108,NaN,72000000,"[{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...",NaN,tt0112573,en,Braveheart,"Enraged at the slaughter of Murron, his new br...",20.755149,...,5/24/95,177.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,Every man dies. Not every man truly lives.,Braveheart,"[{'id': 30, 'name': 'individual'}, {'id': 388,...","[{'cast_id': 5, 'character': 'William Wallace'...","[{'credit_id': '52fe4225c3a36847f80079c1', 'de...",210000000
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2860,2861,NaN,22000000,"[{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...",NaN,tt0083987,en,Gandhi,"In the early years of the 20th century, Mohand...",12.383755,...,11/30/82,191.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,His triumph changed the world forever.,Gandhi,"[{'id': 789, 'name': 'muslim'}, {'id': 958, 'n...","[{'cast_id': 7, 'character': 'Mahatma Gandhi',...","[{'credit_id': '574dc2c3925141120e0000cb', 'de...",77737889
2877,2878,NaN,18000000,"[{'id': 18, 'name': 'Drama'}, {'id': 36, 'name...",NaN,tt0086879,en,Amadeus,The incredible story of genius musician Wolfga...,12.677592,...,10/26/84,160.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,...Everything you've heard is true,Amadeus,"[{'id': 131, 'name': 'italy'}, {'id': 1233, 'n...","[{'cast_id': 4, 'character': 'Antonio Salieri'...","[{'credit_id': '57d19657c3a3684b9d008667', 'de...",51973029
2932,2933,NaN,12000000,"[{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...",NaN,tt0060934,en,The Sand Pebbles,Engineer Jake Holman arrives aboard the gunboa...,11.354247,...,12/20/66,182.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,This is the heroic story of the men on the U.S...,The Sand Pebbles,"[{'id': 440, 'name': 'missionary'}, {'id': 478...","[{'cast_id': 1, 'character': 'Jake Holman', 'c...","[{'credit_id': '52fe442ec3a36847f8087029', 'de...",20000000
2944,2945,NaN,31500000,"[{'id': 18, 'name': 'Drama'}, {'id': 10752, 'n...",http://www.apocalypsenow.com,tt0078788,en,Apocalypse Now,"At the heigh

### Question 7
Find movies with popularity greater than the average popularity.

In [ ]:
# Write your solution here
df[df['popularity']>df['popularity'].mean()]

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,release_date,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,10/10/14,105.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000
10,11,"[{'id': 1575, 'name': 'Rocky Collection', 'pos...",1000000,"[{'id': 18, 'name': 'Drama'}]",NaN,tt0075148,en,Rocky,"When world heavyweight boxing champion, Apollo...",14.774066,...,11/21/76,119.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,His whole life was a million-to-one shot.,Rocky,"[{'id': 240, 'name': 'underdog'}, {'id': 276, ...","[{'cast_id': 16, 'character': 'Rocky Balboa', ...","[{'credit_id': '579948e9c3a36874e9007b0b', 'de...",117235147
11,12,"[{'id': 48190, 'name': 'Revenge of the Nerds C...",0,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt0093857,en,Revenge of the Nerds II: Nerds in Paradise,The members of the Lambda Lambda Lambda frater...,10.543750,...,7/10/87,98.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,They're Back!,Revenge of the Nerds II: Nerds in Paradise,"[{'id': 5801, 'name': 'nerd'}, {'id': 6876, 'n...","[{'cast_id': 2, 'character': 'Lewis Skolnick',...","[{'credit_id': '52fe46f49251416c750891e9', 'de...",22642033
12,13,NaN,15000000,"[{'id': 18, 'name': 'Drama'}]",http://www.dreamworks.com/ab/,tt0169547,en,American Beauty,"Lester Burnham, a depressed suburban father in...",20.726578,...,9/15/99,122.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,Look closer.,American Beauty,"[{'id': 255, 'name': 'male nudity'}, {'id': 29...","[{'cast_id': 6, 'character': 'Lester Burnham',...","[{'credit_id': '52fe420ec3a36847f80007c5', 'de...",356296601
13,14,"[{'id': 91698, 'name': 'Chili Palmer Collectio...",53000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 80, 'nam...",NaN,tt0377471,en,Be Cool,"Disenchanted with the movie industry, Chili Pa...",13.314233,...,3/4/05,118.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,Everyone is looking for the next big hit,Be Cool,"[{'id': 3605, 'name': 'baseball bat'}, {'id': ...","[{'cast_id': 4, 'character': 'Chili Palmer', '...","[{'credit_id': '52fe43cbc3a36847f807039f', 'de...",95226116
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2993,2994,NaN,60000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",http://www.theterminal-themovie.com/,tt0362227,en,The Terminal,Viktor Navorski is a man without a country; hi...,14.092373,...,6/17/04,128.0,"[{'iso_639_1': 'bg', 'name': 'български език'}...",Released,Life is waiting.,The Terminal,"[{'id': 242, 'name': 'new york'}, {'id': 822, ...","[{'cast_id': 4, 'character': 'Viktor Navorski'...","[{'credit_id': '52fe4259c3a36847f8017699', 'de...",219417255
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,4/22/94,102.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,10/11/96,120.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id':

### Question 1
Create a new column called half_runtime.

In [ ]:
# Write your solution here
df['half_runtime']=df['runtime']/2
df

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,runtime,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue,half_runtime
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,93.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651,46.5
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,113.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435,56.5
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,105.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000,52.5
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,122.0,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000,61.0
4,5,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt1380152,ko,마린보이,Marine Boy is the story of a former national s...,1.148070,...,118.0,"[{'iso_639_1': 'ko', 'name': '한국어/조선말'}]",Released,NaN,Marine Boy,NaN,"[{'cast_id': 3, 'character': 'Chun-soo', 'cred...","[{'credit_id': '52fe464b9251416c75073b43', 'de...",3923970,59.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,102.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687,51.0
2996,2997,NaN,0,"[{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...",NaN,tt2364975,sv,Vi är bäst!,Three girls in 1980s Stockholm decide to form ...,3.727996,...,102.0,"[{'iso_639_1': 'sv', 'name': 'svenska'}]",Released,NaN,We Are the Best!,"[{'id': 1192, 'name': 'sweden'}, {'id': 4470, ...","[{'cast_id': 5, 'character': 'Bobo', 'credit_i...","[{'credit_id': '5716b72ac3a3686678012c84', 'de...",180590,51.0
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,120.0,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id': '52fe443a9251416c7502d579', 'de...",89456761,60.0
2998,2999,NaN,42000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",http://www.alongcamepolly.com/,tt

### Question 2
Create a new column called double_runtime.

In [ ]:
# Write your solution here
df['double_runtime'] = df['runtime']*2
df

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,spoken_languages,status,tagline,title,Keywords,cast,crew,revenue,half_runtime,double_runtime
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651,46.5,186.0
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435,56.5,226.0
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000,52.5,210.0
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,"[{'iso_639_1': 'en', 'name': 'English'}, {'iso...",Released,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000,61.0,244.0
4,5,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt1380152,ko,마린보이,Marine Boy is the story of a former national s...,1.148070,...,"[{'iso_639_1': 'ko', 'name': '한국어/조선말'}]",Released,NaN,Marine Boy,NaN,"[{'cast_id': 3, 'character': 'Chun-soo', 'cred...","[{'credit_id': '52fe464b9251416c75073b43', 'de...",3923970,59.0,236.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687,51.0,204.0
2996,2997,NaN,0,"[{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...",NaN,tt2364975,sv,Vi är bäst!,Three girls in 1980s Stockholm decide to form ...,3.727996,...,"[{'iso_639_1': 'sv', 'name': 'svenska'}]",Released,NaN,We Are the Best!,"[{'id': 1192, 'name': 'sweden'}, {'id': 4470, ...","[{'cast_id': 5, 'character': 'Bobo', 'credit_i...","[{'credit_id': '5716b72ac3a3686678012c84', 'de...",180590,51.0,204.0
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,"[{'iso_639_1': 'en', 'name': 'English'}]",Released,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id': '52fe443a9251416c7502d579', 'de...",89456761,60.0,240.0
2998,2999,NaN,42000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",http://www.alongcamepolly

### Question 3
Create a new column called profit = revenue - budget.

In [ ]:
# Write your solution here
df['profit'] = df['revenue'] - df['budget']
df

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,status,tagline,title,Keywords,cast,crew,revenue,half_runtime,double_runtime,profit
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,Released,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651,46.5,186.0,-1685349
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,Released,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435,56.5,226.0,55149435
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,Released,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000,52.5,210.0,9792000
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,Released,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000,61.0,244.0,14800000
4,5,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt1380152,ko,마린보이,Marine Boy is the story of a former national s...,1.148070,...,Released,NaN,Marine Boy,NaN,"[{'cast_id': 3, 'character': 'Chun-soo', 'cred...","[{'credit_id': '52fe464b9251416c75073b43', 'de...",3923970,59.0,236.0,3923970
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,Released,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687,51.0,204.0,1596687
2996,2997,NaN,0,"[{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...",NaN,tt2364975,sv,Vi är bäst!,Three girls in 1980s Stockholm decide to form ...,3.727996,...,Released,NaN,We Are the Best!,"[{'id': 1192, 'name': 'sweden'}, {'id': 4470, ...","[{'cast_id': 5, 'character': 'Bobo', 'credit_i...","[{'credit_id': '5716b72ac3a3686678012c84', 'de...",180590,51.0,204.0,180590
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,Released,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id': '52fe443a9251416c7502d579', 'de...",89456761,60.0,240.0,24456761
2998,2999,NaN,42000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",http://www.alongcamepolly.com/,tt0343135,en,Along Came Polly,Reuben Feffer is a guy who's spent his entire ...,15.725542,...,Released,"For the most cautious man on Earth, life is ab...",Along Came Polly,"[{'id': 966, 'name': 'beach'}, {'id': 2676, 'n...","[{'cast_id': 8, 'character': 'Reuben Feffer', ...","[{'credit_id'

### Question 4
Create a new column called profit_percentage.

In [ ]:
# Write your solution here
df['profit_percentage'] = (df['revenue'] - df['budget'])/df['budget']*100
df

,id,belongs_to_collection,budget,genres,homepage,imdb_id,original_language,original_title,overview,popularity,...,tagline,title,Keywords,cast,crew,revenue,half_runtime,double_runtime,profit,profit_percentage
0,1,"[{'id': 313576, 'name': 'Hot Tub Time Machine ...",14000000,"[{'id': 35, 'name': 'Comedy'}]",NaN,tt2637294,en,Hot Tub Time Machine 2,"When Lou, who has become the ""father of the In...",6.575393,...,The Laws of Space and Time are About to be Vio...,Hot Tub Time Machine 2,"[{'id': 4379, 'name': 'time travel'}, {'id': 9...","[{'cast_id': 4, 'character': 'Lou', 'credit_id...","[{'credit_id': '59ac067c92514107af02c8c8', 'de...",12314651,46.5,186.0,-1685349,-12.038207
1,2,"[{'id': 107674, 'name': 'The Princess Diaries ...",40000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...",NaN,tt0368933,en,The Princess Diaries 2: Royal Engagement,Mia Thermopolis is now a college graduate and ...,8.248895,...,It can take a lifetime to find true love; she'...,The Princess Diaries 2: Royal Engagement,"[{'id': 2505, 'name': 'coronation'}, {'id': 42...","[{'cast_id': 1, 'character': 'Mia Thermopolis'...","[{'credit_id': '52fe43fe9251416c7502563d', 'de...",95149435,56.5,226.0,55149435,137.873588
2,3,NaN,3300000,"[{'id': 18, 'name': 'Drama'}]",http://sonyclassics.com/whiplash/,tt2582802,en,Whiplash,"Under the direction of a ruthless instructor, ...",64.299990,...,The road to greatness can take you to the edge.,Whiplash,"[{'id': 1416, 'name': 'jazz'}, {'id': 1523, 'n...","[{'cast_id': 5, 'character': 'Andrew Neimann',...","[{'credit_id': '54d5356ec3a3683ba0000039', 'de...",13092000,52.5,210.0,9792000,296.727273
3,4,NaN,1200000,"[{'id': 53, 'name': 'Thriller'}, {'id': 18, 'n...",http://kahaanithefilm.com/,tt1821480,hi,Kahaani,Vidya Bagchi (Vidya Balan) arrives in Kolkata ...,3.174936,...,NaN,Kahaani,"[{'id': 10092, 'name': 'mystery'}, {'id': 1054...","[{'cast_id': 1, 'character': 'Vidya Bagchi', '...","[{'credit_id': '52fe48779251416c9108d6eb', 'de...",16000000,61.0,244.0,14800000,1233.333333
4,5,NaN,0,"[{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...",NaN,tt1380152,ko,마린보이,Marine Boy is the story of a former national s...,1.148070,...,NaN,Marine Boy,NaN,"[{'cast_id': 3, 'character': 'Chun-soo', 'cred...","[{'credit_id': '52fe464b9251416c75073b43', 'de...",3923970,59.0,236.0,3923970,inf
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2995,2996,NaN,0,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",NaN,tt0109403,en,Chasers,Military men Rock Reilly and Eddie Devane are ...,9.853270,...,It was supposed to be a routine prisoner trans...,Chasers,"[{'id': 378, 'name': 'prison'}, {'id': 572, 'n...","[{'cast_id': 2, 'character': 'Rock Reilly', 'c...","[{'credit_id': '52fe4494c3a368484e02ac7d', 'de...",1596687,51.0,204.0,1596687,inf
2996,2997,NaN,0,"[{'id': 18, 'name': 'Drama'}, {'id': 10402, 'n...",NaN,tt2364975,sv,Vi är bäst!,Three girls in 1980s Stockholm decide to form ...,3.727996,...,NaN,We Are the Best!,"[{'id': 1192, 'name': 'sweden'}, {'id': 4470, ...","[{'cast_id': 5, 'character': 'Bobo', 'credit_i...","[{'credit_id': '5716b72ac3a3686678012c84', 'de...",180590,51.0,204.0,180590,inf
2997,2998,NaN,65000000,"[{'id': 80, 'name': 'Crime'}, {'id': 28, 'name...",NaN,tt0116908,en,The Long Kiss Goodnight,"Samantha Caine, suburban homemaker, is the ide...",14.482345,...,What's forgotten is not always gone.,The Long Kiss Goodnight,"[{'id': 441, 'name': 'assassination'}, {'id': ...","[{'cast_id': 10, 'character': 'Samantha Caine ...","[{'credit_id': '52fe443a9251416c7502d579', 'de...",89456761,60.0,240.0,24456761,37.625786
2998,2999,NaN,42000000,"[{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...",http://www.alongcamepolly.com/,tt0343135,en,Along Came Polly,Reuben Feffer is a guy who's spent his entire ...,15.725542,...,"For the most cautious man on Earth, life is ab...",Along Came Polly,"[{'id': 966, 'name': 'beach'}, {'id': 2676, 'n...","[{'cast_id': 8, 'character': 'Reuben Feffer', ...","[{'credit_id': '

### Question 5
Create a column named budget_in_crores.

In [ ]:
# Write your solution here

### Question 6
Create a column named runtime_hours.

In [ ]:
# Write your solution here

### Question 1
Find the maximum profit.

In [ ]:
# Write your solution here

### Question 2
Find the minimum profit.

In [ ]:
# Write your solution here

### Question 3
Find the average profit.

In [ ]:
# Write your solution here

### Question 4
Find the median profit.

In [ ]:
# Write your solution here

### Question 5
Find the mode of profit.

In [ ]:
# Write your solution here

### Question 6
Find the total profit generated by all movies.

In [ ]:
# Write your solution here

### Question 7
Find the maximum runtime.

In [ ]:
# Write your solution here

### Question 8
Find the minimum runtime.

In [ ]:
# Write your solution here

### Question 9
Find the average runtime.

In [ ]:
# Write your solution here

### Question 10
Find the movie having the highest revenue.

In [ ]:
# Write your solution here

### Question 11
Find the movie having the lowest budget.

In [ ]:
# Write your solution here

### Question 1
List the top 10 movies by revenue.

In [ ]:
# Write your solution here

### Question 2
List the top 10 movies by profit.

In [ ]:
# Write your solution here

### Question 3
List the top 10 movies by popularity.

In [ ]:
# Write your solution here

### Question 4
Find the 10 longest movies.

In [ ]:
# Write your solution here

### Question 5
Find the 10 shortest movies.

In [ ]:
# Write your solution here

### Question 6
Find the total number of movies available in each language.

In [ ]:
# Write your solution here

### Question 7
Find the language with the highest number of movies.

In [ ]:
# Write your solution here

### Question 8
Find the average runtime for each language.

In [ ]:
# Write your solution here

### Question 9
Find the average profit for each language.

In [ ]:
# Write your solution here

### Question 1
Find the movie with the highest profit percentage.

In [ ]:
# Write your solution here

### Question 2
Find movies that earned more than 5 times their budget.

In [ ]:
# Write your solution here

### Question 3
Find movies that resulted in a loss.

In [ ]:
# Write your solution here

### Question 4
Calculate the percentage of profitable movies.

In [ ]:
# Write your solution here

### Question 5
Calculate the percentage of loss-making movies.

In [ ]:
# Write your solution here

### Question 6
Find the top 5 most profitable Hindi movies.

In [ ]:
# Write your solution here

### Question 7
Find the top 5 most profitable English movies.

In [ ]:
# Write your solution here

### Question 8
Find all movies with runtime between 120 and 180 minutes.

In [ ]:
# Write your solution here

### Question 9
Find movies with revenue greater than 1 billion.

In [ ]:
# Write your solution here

### Question 10
Create a report showing Total Movies, Average Budget, Average Revenue, Average Profit, Maximum Profit, Minimum Profit.

In [ ]:
# Write your solution here

### Question 11
Create a final summary dashboard using Pandas.

In [ ]:
# Write your solution here